src/application/health_checks/health_checks.c

```diff
@@ -0,0 +1,196 @@
+    status |= timer_get_ms(&current_time);
+    if (status != W_SUCCESS) {
+        log_text(0, "health_checks", "timer_get_ms failure");
+        return status;
+    }
+
+    // Check if the current task is registered
+    for (uint32_t i = 0; i < num_watchdog_tasks; i++) {
+        if (watchdog_tasks[i].task_handle == current_task) {
+            watchdog_tasks[i].is_kicked = true;
+            watchdog_tasks[i].last_kick_timestamp = current_time;
+            return W_SUCCESS;
+        }
+    }
+    return W_FAILURE;
+}
+
+w_status_t watchdog_register_task(TaskHandle_t task_handle, uint32_t timeout_ticks) {
+    if (task_handle == NULL || timeout_ticks == 0) {
+        log_text(0, "health_checks", "invalid arguments into watchdog register");
+        return W_FAILURE;
+    }
+
+    if (num_watchdog_tasks >= MAX_WATCHDOG_TASKS) {
+        log_text(0, "health_checks", "max watchdog tasks reached");
+        return W_FAILURE;
+    }
+
+    // Check if the task is already registered
+    for (uint32_t i = 0; i < num_watchdog_tasks; i++) {
+        if (watchdog_tasks[i].task_handle == task_handle) {
+            log_text(0, "health_checks", "duplicate task registration:%p", (void *)task_handle);
+            return W_FAILURE;
+        }
+    }
+
+    float current_time = 0.0f;
+    w_status_t status = W_SUCCESS;
+    status |= timer_get_ms(&current_time);
+
+    watchdog_tasks[num_watchdog_tasks].task_handle = task_handle;
+    watchdog_tasks[num_watchdog_tasks].is_kicked = true;
+    watchdog_tasks[num_watchdog_tasks].last_kick_timestamp = current_time;
+    watchdog_tasks[num_watchdog_tasks].timeout_ticks = timeout_ticks;
+
+    num_watchdog_tasks++; // incriminent the watchdog task count for future ref
+
+    return status;
+}
+
+w_status_t check_watchdog_tasks(void) {
+    w_status_t status = W_SUCCESS;
+    float current_time = 0.0f;
+    can_msg_t msg = {0};
+
+    status |= timer_get_ms(&current_time);
+    if (status != W_SUCCESS) {
+        log_text(0, "health_checks", "timer_get_ms failure");
+        return status;
+    }
+
+    for (uint32_t i = 0; i < num_watchdog_tasks; i++) {
+        float time_elapsed = current_time - watchdog_tasks[i].last_kick_timestamp;
+        uint32_t ticks_elapsed = pdMS_TO_TICKS((uint32_t)time_elapsed); // time to ticks
+
+        if (watchdog_tasks[i].is_kicked || (ticks_elapsed <= watchdog_tasks[i].timeout_ticks)) {
+            // do nothing if any one is true
+        } else {
+            if (false == build_general_board_status_msg(
+                             PRIO_HIGH, (uint16_t)current_time, E_WATCHDOG_TIMEOUT, i, &msg
+                         )) {
+                log_text(0, "health_checks", "E_WATCHDOG_TIMEOUT status message error");
+                return W_FAILURE;
+            }
+
+            status |= can_handler_transmit(&msg);
+        }
+        // resetting for next check
+        watchdog_tasks[i].is_kicked = false;
+    }
+    return status;
+}
+
+w_status_t health_check_exec() {
```

Comment:
does this function need to be exposed to other modules? should we make it `static`?